cedar-lean/Cedar/Thm/Data/List.lean

```diff
@@ -114,70 +114,9 @@ theorem filterMap_equiv (f : α → Option β) (xs ys : List α) :
   apply And.intro <;>
   intro b a h₃ h₄ <;>
   exists a <;>
-  simp [h₄]
-  exact h₁ h₃
-  exact h₂ h₃
-
-theorem filterMap_empty_iff_f_returns_none {f : α → Option β} {xs : List α} :
-  xs.filterMap f = [] ↔ ∀ x ∈ xs, f x = none
-:= by
-  constructor
-  case mp =>
-    induction xs
-    case nil => simp
-    case cons x xs h_ind =>
-      intro h₁ a h₂
-      simp only [List.filterMap_cons] at h₁
-      split at h₁ <;> try simp at h₁
-      case h_1 h₃ =>
-        rcases (List.mem_cons.mp h₂) with h₄ | h₄
-        case inl => subst h₄ ; assumption
-        case inr => apply h_ind h₁ a ; assumption
-  case mpr =>
-    intro h₁
-    induction xs
-    case nil => simp
-    case cons x xs h_ind =>
-      simp only [List.filterMap_cons]
-      split
-      case h_1 =>
-        apply h_ind ; clear h_ind
-        intro a h₂
-        apply h₁ a
-        exact List.mem_cons_of_mem x h₂
-      case h_2 b h₂ =>
-        exfalso
-        specialize h₁ x
-        simp at h₁
-        simp [h₁] at h₂
-
-theorem filterMap_nonempty_iff_exists_f_returns_some {f : α → Option β} {xs : List α} :
-  xs.filterMap f ≠ [] ↔ ∃ x ∈ xs, (f x).isSome
```

Comment:
Moved to a dedicated section for `filterMap`, and renamed `filterMap_nonempty_iff_exists`

Reply:
Conversation moved to https://github.com/cedar-policy/cedar-spec/pull/305#discussion_r1587727061